lib/puppet/provider/group/windows_adsi.rb

```diff
@@ -72,11 +72,12 @@ def exists?
 
   def delete
     Puppet::Util::Windows::ADSI::Group.delete(@resource[:name])
+    @group.deleted=true if @group
```

Comment:
I think it might be easier to keep the deleted flag in the provider and not on the adsi object, e.g. `@deleted = true`, and then down below:

```
@group.commit if @group && !@deleted
```